src/core/qgis.h

```diff
@@ -2001,6 +2001,7 @@ class CORE_EXPORT Qgis
       DontStoreOriginalStyles SIP_MONKEYPATCH_COMPAT_NAME( FlagDontStoreOriginalStyles ) = 1 << 3, //!< Skip the initial XML style storage for layers. Useful for minimising project load times in non-interactive contexts.
       DontLoad3DViews SIP_MONKEYPATCH_COMPAT_NAME( FlagDontLoad3DViews ) = 1 << 4, //!< Skip loading 3D views (since QGIS 3.26)
       DontLoadProjectStyles = 1 << 5, //!< Skip loading project style databases (deprecated -- use ProjectCapability::ProjectStyles flag instead)
+      ForceLayerReadOnly SIP_MONKEYPATCH_COMPAT_NAME( FlagForceLayerReadOnly ) = 1 << 6, //!< Open layers in a read-only mode. (since QGIS 3.28)
```

Comment:
```suggestion
      ForceLayerReadOnly = 1 << 6, //!< Open layers in a read-only mode. (since QGIS 3.28)
```

We only need the compat_name when maintaining existing API is a factor -- here it's a completely new flag so isn't required.